src/uct/tcp/tcp_listener.c

```diff
@@ -171,11 +172,29 @@ ucs_status_t uct_tcp_listener_reject(uct_listener_h listener,
               cep, cep->fd, cep->state,
               uct_tcp_sockcm_cm_ep_peer_addr_str(cep, peer_str, UCS_SOCKADDR_STRING_LEN));
 
-    status = UCS_OK;
+    hdr = (uct_tcp_sockcm_priv_data_hdr_t*)cep->comm_ctx.buf;
+
+    hdr->length          = 0;   /* sending only the header in the reject message */
+    hdr->status          = (uint8_t)UCS_ERR_REJECTED;
+    cep->comm_ctx.length = sizeof(*hdr);
+
+    /* the server needs to send a reject message to the client */
+    ucs_async_modify_handler(cep->fd, UCS_EVENT_SET_EVWRITE | UCS_EVENT_SET_EVREAD);
```

Comment:
why need EVREAD?

Reply:
in case the client closes its ep, the server shouldn't miss a disconnect (it has already called accept())

Reply:
how could it call accept if we call reject here?

Reply:
if you mean tcp socket accept, it doesn't matter, after calling reject the server should not get a disconnect callback

Reply:
yes i meant tcp socket accept.
what about disconnects during the sending of the reject message? 
or you mean it's ok to miss disconnects once the server decided to reject?

Reply:
yes, after we reject we don't care if remote disconnected. we are not going to forward the disconnect to UCP anyway.